internal/api/validate/validate.go

```diff
@@ -109,8 +110,11 @@ func validateVariableNames(variables []dashboard.Variable) error {
 }
 
 func validateDatasourcePlugin(plugin common.Plugin, name string, sch schema.Schema) error {
-	if _, err := http.ValidateAndExtract(plugin.Spec); err != nil {
```

Comment:
if `ValidateAndExtract` is still returning the config + kind like I have suggested in a different comment, then you don't need to change this function.